src/uiRouterNgModule.ts

```diff
@@ -16,12 +16,24 @@ import { ROUTES } from "@angular/router";
 /** @hidden */ export const UIROUTER_STATES       = new OpaqueToken("UIRouter States");
 // /** @hidden */ export const ROUTES = UIROUTER_STATES;
 
+export function onTransitionReady(transitionService: TransitionService, plateformId) {
+    if (isPlatformServer(plateformId)) {
+       return () => Promise.resolve();
+    }
+
+    return () => new Promise(resolve => {
+      transitionService.onSuccess({}, resolve);
```

Comment:
I think it should work, there is something wrong in my app that prevent it and I didn't have the time to deep into it right now